tests/processors/base/writers/test_trial_and_record.py

```diff
@@ -284,16 +284,23 @@ def test_writes_record_with_valid_url(self, conn, trial, nct_source):
 
         assert conn['database']['records'].count(id=record_id) == 1
 
-    def test_doesnt_update_None_parameters(self, conn, trial, record):
+    def test_doesnt_alter_record_parameters(self, conn, trial, record):
         trial_attrs = conn['database']['trials'].find_one(id=trial)
-        record_attrs = conn['database']['records'].find_one(id=record)
+        conn_mock = mock.MagicMock()
+        conn_mock['database']['records'].find_one.return_value = None
+
+        with pytest.raises(ValueError):
+            trial_and_record_writer._write_record(conn_mock, trial_attrs, record, None, None, False)
+
+            conn_mock['database']['records'].upsert.assert_called()
 
-        trial_and_record_writer._write_record(conn, trial_attrs, record, None, None)
+            upsert_call_args = conn_mock['database']['records'].upsert.call_args
+            upsertted_record = upsert_call_args[0][0]
+            assert upsertted_record['is_primary'] is False
+            assert upsertted_record['source_url'] is None
+            assert upsertted_record['source_id'] is None
```

Comment:
These assertions never run, because `_write_record()` does raise an Exception. See:

```python
# This test pass
def test_foo():
    with pytest.raises(Exception):
        raise Exception()
        assert False  # This never gets executed

    assert True  # This does
```